src/storage/hummock_sdk/src/version.rs

```diff
@@ -0,0 +1,184 @@
+    pub fn from_persisted_protobuf(pb_version: &PbHummockVersion) -> Self {
+        Self::from_protobuf_inner(pb_version)
+    }
+
+    fn from_protobuf_inner(pb_version: &PbHummockVersion) -> Self {
+        Self {
+            id: pb_version.id,
+            levels: pb_version
+                .levels
+                .iter()
+                .map(|(group_id, levels)| (*group_id as CompactionGroupId, levels.clone()))
+                .collect(),
+            max_committed_epoch: pb_version.max_committed_epoch,
+            safe_epoch: pb_version.safe_epoch,
+            table_watermarks: pb_version
+                .table_watermarks
+                .iter()
+                .map(|(table_id, table_watermark)| {
+                    (
+                        TableId::new(*table_id),
+                        TableWatermarks::from_protobuf(table_watermark),
+                    )
+                })
+                .collect(),
+        }
+    }
+
+    pub fn to_protobuf(&self) -> PbHummockVersion {
+        PbHummockVersion {
+            id: self.id,
+            levels: self
+                .levels
+                .iter()
+                .map(|(group_id, levels)| (*group_id as _, levels.clone()))
+                .collect(),
+            max_committed_epoch: self.max_committed_epoch,
+            safe_epoch: self.safe_epoch,
+            table_watermarks: self
+                .table_watermarks
+                .iter()
+                .map(|(table_id, watermark)| (table_id.table_id, watermark.to_protobuf()))
+                .collect(),
+        }
+    }
+
+    pub fn estimated_encode_len(&self) -> usize {
+        self.levels.len() * size_of::<CompactionGroupId>()
+            + self
+                .levels
+                .values()
+                .map(|level| level.encoded_len())
+                .sum::<usize>()
+            + self.table_watermarks.len() * size_of::<u32>()
+            + self
+                .table_watermarks
+                .values()
+                .map(|table_watermark| table_watermark.estimated_encode_len())
+                .sum::<usize>()
+    }
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct HummockVersionDelta {
+    pub id: u64,
+    pub prev_id: u64,
+    pub group_deltas: HashMap<CompactionGroupId, PbGroupDeltas>,
+    pub max_committed_epoch: u64,
+    pub safe_epoch: u64,
+    pub trivial_move: bool,
+    pub gc_object_ids: Vec<HummockSstableObjectId>,
+    pub new_table_watermarks: HashMap<TableId, TableWatermarks>,
+    pub removed_table_ids: Vec<TableId>,
+}
+
+impl Default for HummockVersionDelta {
+    fn default() -> Self {
+        HummockVersionDelta::from_protobuf_inner(&PbHummockVersionDelta::default())
+    }
+}
+
+impl HummockVersionDelta {
+    pub fn from_persisted_protobuf(delta: &PbHummockVersionDelta) -> Self {
+        Self::from_protobuf_inner(delta)
+    }
+
+    pub fn from_rpc_protobuf(delta: &PbHummockVersionDelta) -> Self {
+        Self::from_protobuf_inner(delta)
+    }
```

Comment:
Need some docs to explain why we need to distinguish rpc protobuf and persisted protobuf.